main_test.go

```diff
@@ -115,32 +115,69 @@ func TestParseFlagsError(t *testing.T) {
 type processGeofeedTest struct {
 	gf string
 	db string
-	dl string
+	dl []string
 	c  counts
+	em string
 }
 
 func TestProcessGeofeed(t *testing.T) {
-	tests := []processGeofeedTest{
+	goodTests := []processGeofeedTest{
 		{
 			"test_data/geofeed1.csv",
 			"test_data/GeoIP2-City-Test.mmdb",
-			"Found a potential improvement: '2a02:ecc0::/29",
+			[]string{
+				"Found a potential improvement: '2a02:ecc0::/29",
+				"Found a potential improvement: '202.196.224.5/32",
+			},
 			counts{
-				1,
-				1,
+				2,
+				2,
 			},
+			"",
 		},
 	}
+
 	// Testing the full content of the difference explanation strings is likely to be
 	// tedious and brittle, so we will just check for some substring.
-	for _, test := range tests {
+	for _, test := range goodTests {
 		t.Run(
 			strings.Join([]string{test.gf, test.db}, " "), func(t *testing.T) {
-				c, diffLines, err := processGeofeed(test.gf, test.db)
+				c, dl, err := processGeofeed(test.gf, test.db)
 				assert.NoError(t, err, "processGeofeed ran without error")
-				assert.Contains(t, diffLines[0], test.dl, "got expected substring")
+				for i, s := range test.dl {
+					assert.Contains(
+						t,
+						dl[i],
+						s,
+						"got expected substring: '%s', substring",
+					)
+				}
 				assert.Equal(t, test.c, c, "processGeofeed returned expected results")
 			},
 		)
 	}
+
+	badTests := []processGeofeedTest{
+		{
+			"test_data/geofeed-missing-fields.csv",
+			"test_data/GeoIP2-City-Test.mmdb",
+			[]string{},
+			counts{},
+			"saw fewer than the expected 5 fields at line 1",
+		},
+	}
+
+	for _, test := range badTests {
+		t.Run(
+			strings.Join([]string{test.gf, test.db}, " "), func(t *testing.T) {
+				_, _, err := processGeofeed(test.gf, test.db)
+				assert.Errorf(
```

Comment:
If I print out err here I see `open test_data/geofeed-missing-fields.csv: no such file or directory`. Not "saw fewer than the expected 5 fields at line 1".

Reply:
Yeah, that assertion just looks if there is any error, while I really wanted to check for a specific error, which dec98805bd149bbb857e8d3f52f5474ddda38d7b fixes.